doc/source/cluster/cluster_under_construction/key-concepts.rst

```diff
@@ -1,130 +1,83 @@
+singleton processes responsible for cluster management such as the :ref:`autoscaler<cluster-autoscaler-under-construction>`
+and the Ray driver processes, :ref:`which run the top-level Ray application
+<cluster-clients-and-jobs-under-construction>`. Ray may schedule tasks and actors
+on the head node just like any other worker node.
+
+.. note::
+    :ref:`On Kubernetes<kuberay-index>`, the autoscaler process runs inside a sidecar container in the head pod,
+    instead of in the same container (which would be most analogous to how the autoscaler runs inside the head node
+    on non-Kubernetes clusters).
 
 .. _cluster-autoscaler-under-construction:
 
 Autoscaler
 ----------
 
-The autoscaler is a process that runs on the :ref:`head node<cluster-head-node>`
-and is responsible for adding or removing :ref:`worker nodes<cluster-worker-node>`
-to meet the needs of the Ray workload while matching the specification in the
-:ref:`cluster config file<cluster-config>`. In particular, if the resource
-demands of the Ray workload exceed the current capacity of the cluster, the
-autoscaler will try to add nodes. Conversely, if a node is idle for long enough,
-the autoscaler will remove it from the cluster. To learn more about autoscaling,
-refer to the :ref:`Ray cluster deployment guide<deployment-guide-autoscaler>`.
-
-Ray Client
-----------
-The Ray Client is an API that connects a Python script to a remote Ray cluster.
-To learn more about the Ray Client, you can refer to the :ref:`documentation<ray-client>`.
-
-Job submission
---------------
+The autoscaler is a process that runs on the :ref:`head node<cluster-head-node-under-construction>`.
+It is responsible for provisioning or deprovisioning :ref:`worker nodes<cluster-worker-nodes-under-construction>`
+to meet the needs of the Ray workload. In particular, if the resource demands of the Ray workload exceed the
+current capacity of the cluster, the autoscaler will attempt to add more nodes. Conversely, if
+a node is idle for long enough, the autoscaler will remove it from the cluster.
 
-Ray Job submission is a mechanism to submit locally developed and tested applications
-to a remote Ray cluster. It simplifies the experience of packaging, deploying,
-and managing a Ray application. To learn more about Ray jobs, refer to the
-:ref:`documentation<ray-job-submission-api-ref>`.
+To learn more about the autoscaler and how to configure it, refer to the following user guides:
 
-Cloud clusters
---------------
+* :ref:`Configuring Autoscaling on VMs<deployment-guide-autoscaler-under-construction>`.
+* :ref:`Configuring Autoscaling on Kubernetes<kuberay-index>` (TODO cade@ update this link.)
 
-If you’re using AWS, GCP, Azure (community-maintained) or Aliyun (community-maintained), you can use the
-:ref:`Ray cluster launcher<cluster-launcher>` to launch cloud clusters, which
-greatly simplifies the cluster setup process.
+.. _cluster-clients-and-jobs-under-construction:
 
-Cluster managers
+Clients and Jobs
```

Comment:
This section is not very accurate after we consolidate the concept of ray jobs (see [design doc](https://docs.google.com/document/d/1Odz5cpev-dGC96fkmos5v08AIs2PTk6sJh_mfmRx1Gs/edit#) here).

Ray jobs include driver jobs and submission jobs.
- Driver jobs are created via ray client or running ray scripts locally
- Submission jobs are created via submission APIs

-------- Here is how it might look like -------
**Ray jobs**

Ray supports two types of jobs for running workloads on a Ray cluster: Driver jobs and Submission jobs.

* **Driver jobs** are created via ray client, which enables interactive development by connecting a local Python script or shell to the cluster....

* **Submission jobs** are created via submission APIs, which enables users to submit locally developed-and-tested applications to a remote Ray...


@alanwguo has more insights into this. Could you review this part?

Reply:
I sync'd with @alanwguo , will update names!